src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -447,11 +449,18 @@ class boss_paletress : public CreatureScript
         void JustSummoned(Creature* summon) override
         {
             argent_challenge_baseAI::JustSummoned(summon);
-            
-            ObjectGuid _memoryGuid = summon->GetGUID();
-            _memory = ObjectAccessor::GetCreature(*me, _memoryGuid);
-            me->GetMotionMaster()->MoveFollow(summon, 30.0f, 0.0f);
-            events.ScheduleEvent(EVENT_MEMORY_AGGRESSIVE, 3s);
+            // does it need a check to check if summoned creature is memory or is that never going to happen?
+            _memoryGuid = summon->GetGUID();
+            Creature* _memory = ObjectAccessor::GetCreature(*me, _memoryGuid);
```

Comment:
If one wants to check if the summon if a particular npc, usually one can check summon->GetEntry().
You can remove the "ObjectAccessor::GetCreature(*me, _memoryGuid);" bit and just use summon instead of _memory as they are the same (and it cannot be null)